anchor/network/src/handshake/record/record.rs

```diff
@@ -0,0 +1,13 @@
+use std::error::Error;
+
+/// The `Record` trait parallels the idea in Go that each record knows how to:
+///   - Provide a domain (for signing separation)
+///   - Provide a "codec" (payload type)
+///   - Marshal to bytes, unmarshal from bytes
+pub trait Record {
```

Comment:
this seems necessary right? As we are only using `NodeInfo`

Reply:
I'll probably remove it.